zero_bin/rpc/src/main.rs

```diff
@@ -1,94 +1,168 @@
 use std::env;
 use std::sync::Arc;
 
+use alloy::primitives::B256;
+use alloy::providers::Provider;
 use alloy::rpc::types::eth::BlockId;
 use alloy::rpc::types::{BlockNumberOrTag, BlockTransactionsKind};
-use clap::{Parser, ValueHint};
+use alloy::transports::Transport;
+use clap::{Args, Parser, Subcommand, ValueHint};
+use evm_arithmetization::GenerationInputs;
 use futures::StreamExt;
+use prover::BlockProverInput;
 use rpc::provider::CachedProvider;
 use rpc::{retry::build_http_retry_provider, RpcType};
 use tracing_subscriber::{prelude::*, EnvFilter};
 use url::Url;
 use zero_bin_common::version;
 use zero_bin_common::{block_interval::BlockInterval, prover_state::persistence::CIRCUIT_VERSION};
 
+#[derive(Args, Clone, Debug)]
+pub(crate) struct Params {
+    // Starting block of interval to fetch
+    #[arg(short, long)]
+    start_block: u64,
+    // End block of interval to fetch
+    #[arg(short, long)]
+    end_block: u64,
+    /// The RPC URL.
```

Comment:
mostly nit: `//` vs `///` and one comment has a final `.` whereas others don't. Would like some unification :)